Latex/samplepaper.tex

```diff
@@ -0,0 +1,288 @@
+\item[Komposit]
+Das Komposit ist so umgesetzt, sodass Komplexe Strukturen wie zum Beispiel Einkaufszentren dargestellt werden können.
+Dafür wird die abstrakte Klasse SensorBereich (SensorArea) genutzt. Diese besitzt die Methode zur Berechnung der Auslastung, dazu muss jeweils die Auslastung der darunter liegenden Läden(Shop) ebenfalls berechnet werden.
+Ein Laden besteht dabei aus mindestens einem Sensorbereich, wobei dieser aus weiteren Läden besteht oder aus konkreten Sensoren(Sensor).
+Dadurch kann der Sensorbereich beliebig skaliert und die eigentliche Auslastung rekursiv berechnet werden.  
+Diese Skalierbarkeit ist der Vorteil dieses Entwurfsmusters und wurde deswegen ausgewählt. 
+\item[Beobachter-Muster]
+Für das Beobachter-Muster wurde zunächst die Klasse MqttService modelliert. 
+Dieser nimmt Sensordaten vom Mqtt-Sensordaten-Server entgegen.
+Damit diese Daten in das richtige Schema der Datenbank persistiert werden können, bietet der Service die Möglichkeit sich darauf einzutragen (subscribe).
+Eingetragene Beobachter (Observer) werden über eingehende Sensordaten informiert.
+Für die Beobachter wird eine Schnittstelle (Interface) bereit gestellt.
+Diese Schnittstelle wird von einer konkreten Klasse implementiert, dem SensorDataWorker.
+Dieser SensorDataWorker filtert die Daten, ruft den DatenbankService auf und übergibt die relevanten Daten.
+Durch dieses Beobachter-Muster ist es möglich weitere Daten über separate Beobachter zu verarbeiten, ohne das bisherige Beobachter angepasst werden müssen.
+\end{description}
+
+
+\subsection{Begründung}
+Der Vorteil des Kompositum-Musters ist, dass der Sensorenbereich (SensorArea) beliebig erweitert werden kann und es dabei zweitrangig ist um was die Shops beziehungsweise andere Komponenten erweitert werden. 
+Ein Nachteil dabei ist, dass die Tiefe ebenfalls beliebig wächst und es so zu einer sehr unübersichtlichen Datenstruktur kommen kann.
+Eine Alternative beziehungsweise Erweiterung wäre das Fabrikmethode-Muster. 
+Dieses Muster erlaubt es uns auf Grundlage der abstrakten Klasse SensorArea weitere Bereiche zu implementieren, ohne dass davon andere Services betroffen sind. 
+Man könnte also Shop oder Einkaufszentrum erzeugen, würde dabei aber die Möglichkeit der Schachtelung verlieren. 
+Um beide Entwurfsmuster möglichst effektiv zu verwenden, wäre eine Kombination denkbar.
+\\
+
+
+Das Beobachter-Muster wurde gewählt, da es eingehende Signale von den Sensoren beziehungsweise Sensornachrichten automatisch verarbeitet und speichert.
+Dieses Muster könnte durch das Zustands-Muster erweitert, aber nicht ersetzt werden.
+Das Zustands-Muster würde in der konkreten Implementierung ebenfalls auf eingehende Nachrichten warten / reagieren, es könnte beispielsweise als ein Buffer agieren welcher Nachrichten sammelt und erst auf Anfrage Daten weiter leitet / speichert.
+Dieses Verhalten würde das Beobachter-Muster eher erweitern oder imitieren und ist daher ungeeignet in diesem Kontext.
```

Comment:
Oben sagst du, das Beobachter-Muster könnte durch das Status-Muster erweitert werden, hier schließt du es aus genau den Gründen aus, warum ziehst du es dann überhaupt in Erwägung?